website/static/js/citationList.js

```diff
@@ -19,10 +22,209 @@ var formatCitation = function(style, data, format) {
+                            var index = parent.citations.indexOf(self);
+                            parent.citations.splice(index, 1);
+                            if (parent.editing()) {
+                                parent.editing(false);
+                            }
+                        }).fail(function(response){
+                            $osf.growl('Error:',
+                                'An unexpected error has occurred.  Please try again later.  If problem persists contact <a href="mailto: support@cos.io">support@cos.io</a>', 'danger'
+                            );
+                        });
+                    }
+                },
+                buttons:{
+                    confirm:{
+                        label:'Delete',
+                        className:'btn-danger'
+                    }
+                }
+            });
+        }
+    };
+
+    self.cancel = function(parent) {
+        if (self.originalValues === undefined) {
+            var index = parent.citations.indexOf(self);
+            parent.citations.splice(index, 1);
+        }
+        else {
+            self.name(self.originalValues.name);
+            self.text(self.originalValues.text);
+            self.view('view');
+        }
+        parent.editing(false);
+        self.messages([]);
+    };
+
+    self.save = function(parent) {
+        if (self.isValid()) {
+            self.messages([]);
+            if (self.messages().length === 0) {
+                var url = $osf.apiV2Url('nodes/' + ctx.node.id + '/citations/');
+                var payload = {};
+                var method = 'POST';
+                payload.data = {
+                    'type': 'citations',
+                    'attributes': {
+                        'name': self.name(),
+                        'text': self.text()
+                    }
+                };
+                if (self.id !== undefined) {
+                    url += self.id + '/';
+
+                    payload.data.attributes.id = self.id;
+                    method = 'PUT';
+                }
+                $osf.ajaxJSON(method, url, {data: payload, isCors: true}).done(function (response) {
+                    var data = response.data;
+                    parent.editing(false);
+                    self.view('view');
+                    self.id = data.id;
+                    if (self.originalValues !== undefined) {
+                        self.originalValues.name = self.name();
+                        self.originalValues.text = self.text();
+                    }
+                    else {
+                        self.originalValues = {
+                            name: self.name(),
+                            text: self.text()
+                        };
+                    }
+                }).fail(function (response) {
+                    if (response.status === 400) {
```

Comment:
@samanehsan The thinking is that this is a bad request so the user should be informed that something was wrong on their end vs our end (400 vs 500) but based on the current implementation I think 400 errors should not happen so we can probably scrap it.